### ISSUE TITLE
Provide better error message with "Required attribute is not set"

### ISSUE THREAD

**Issue by clstokes** [pulumi/pulumi#5674]
#### Problem

The error message "Required attribute is not set" is not clear on the specific attribute that is missing.

#### Error output

```
(venv) p-py-asg % pulumi pre    
Previewing update (dev)

View Live: https://app.pulumi.com/clstokes/p-py-asg/dev/previews/dcc9c7a4-d2b0-4c67-bf81-5a2a2221448f

     Type                      Name          Plan       Info
 +   pulumi:pulumi:Stack       p-py-asg-dev  create     
     └─ aws:autoscaling:Group  repro                    1 error
 
Diagnostics:
  aws:autoscaling:Group (repro):
    error: aws:autoscaling/group:Group resource 'repro' has a problem: Required attribute is not set
```

#### Suggestions for a fix

Provide a more detailed error message. For example:

```
  aws:autoscaling:Group (repro):
    error: aws:autoscaling/group:Group resource 'repro' has a problem: Required attribute is not set: `tags#value` is required
```

#### Steps to reproduce

```
git clone https://gist.github.com/c78dc8f387a8b37155eb31f404eb1f91.git
cd c78dc8f387a8b37155eb31f404eb1f91
pulumi stack init dev
pulumi pre
```

**Comment by clstokes** [comment]
Possibly related and/or in the same area of the code? https://github.com/pulumi/pulumi/issues/5586

**Comment by clstokes** [comment]
Also, slightly related to https://github.com/pulumi/pulumi-terraform-bridge/issues/28 per Luke.

**Comment by leezen** [comment]
I think this was addressed in pulumi/pulumi-terraform-bridge#311. If folks on the issue disagree, please comment and we can re-examine.
